helm-chart/templates/cvat_kvrocks/statefulset.yml

```diff
@@ -0,0 +1,107 @@
+{{- if .Values.cvat.kvrocks.enabled }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ .Release.Name }}-kvrocks
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "cvat.labels" . | nindent 4 }}
+    app: cvat-app
+    tier: kvrocks
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "cvat.labels" . | nindent 6 }}
+      app: cvat-app
+      tier: kvrocks
+  template:
+    metadata:
+      labels:
+        {{- include "cvat.labels" . | nindent 8 }}
+        app: cvat-app
+        tier: kvrocks
+        {{- with .Values.cvat.kvrocks.labels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+      {{- with .Values.cvat.kvrocks.annotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+      containers:
+        - name: cvat-kvrocks-app-container
+          image: {{ .Values.cvat.kvrocks.image }}:{{ .Values.cvat.kvrocks.tag }}
+          imagePullPolicy: {{ .Values.cvat.kvrocks.imagePullPolicy }}
+          args:
+            - --dir
+            - /var/lib/kvrocks/data
+            - --requirepass
+            - "$(CVAT_REDIS_ONDISK_PASSWORD)"
+          {{- with .Values.cvat.kvrocks.resources }}
+          resources:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          ports:
+          - containerPort: 6666
+          env:
+          - name: CVAT_REDIS_ONDISK_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                {{- if .Values.cvat.kvrocks.secret.create }}
+                name: "{{ tpl (.Values.cvat.kvrocks.secret.name) . }}"
+                {{- else }}
+                name: "{{ tpl (.Values.cvat.kvrocks.existingSecret) . }}"
+                {{- end }}
+                key: password
+          {{- with .Values.cvat.kvrocks.additionalEnv }}
+          {{- toYaml . | nindent 10 }}
+          {{- end }}
+          volumeMounts:
+          - name: {{ .Release.Name }}-kvrocks-data
+            mountPath: /var/lib/kvrocks/data
+          {{- with .Values.cvat.kvrocks.additionalVolumeMounts }}
+          {{- toYaml . | nindent 10 }}
+          {{- end }}
+      {{- with .Values.cvat.kvrocks.additionalVolumes }}
+      volumes:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.cvat.kvrocks.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.cvat.kvrocks.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+  {{- if .Values.cvat.kvrocks.defaultStorage.enabled }}
+  volumeClaimTemplates:
+  - metadata:
+      name: {{ .Release.Name }}-kvrocks-data
+      namespace: {{ .Release.Namespace }}
+      annotations:
+        helm.sh/resource-policy: keep
```

Comment:
This annotation isn't needed because Helm doesn't manage volumes created by a StatefulSet. Kubernetes itself does, and by default it will retain the volume when the pod is deleted. So you don't need to do anything special to retain the data between deployments.

Reply:
removed